models/account.py

```diff
@@ -124,7 +124,7 @@ def get_magic_link(
                 ],  # noqa
                 NotifyConstants.MAGIC_LINK_FUND_NAME_FIELD: fund.name,
             }
-            if fund_id and round_id and new_account:
+            if fund_id and round_id and new_account and Config.CREATE_APPLICATION_ON_ACCOUNT_CREATION:
```

Comment:
Should we add some unit tests around this? That might have highlighted the change in behaviour when adding a default value.

Reply:
Yes! Added